Cython/Compiler/ExprNodes.py

```diff
@@ -12213,6 +12213,35 @@ def c_types_okay(self, type1, type2):
             and (type2.is_int or type2.is_enum)
 
 
+class BitwiseOrNode(IntBinopNode):
+    #  '|' operator.
+
+    def analyse_pytyping_modifiers(self, env):
+        if self.operand1.is_none or self.operand2.is_none:
+            return ['typing.Optional']
+
+    def _analyse_bitwise_or_none(self, env, operand_node, leftop_is_none):
+        """Analyse annotations in form `[...] | None` and `None | [...]`"""
+        ttype = operand_node.analyse_as_type(env)
+        if not ttype:
+            return
```

Comment:
I know that `return` is technically the same as `return None` but it's not semantically the same for readers.
```suggestion
        if not ttype:
            return None
```